src/ResourceManager/EventGrid/Commands.EventGrid.Test/ScenarioTests/TopicTests.ps1

```diff
@@ -226,3 +226,168 @@ function TopicNewKeyTests {
+    }
+    catch
+    {
+        Assert-True {$true}
+    }
+
+    try
+    {
+        Write-Debug "Creating a fifth EventGrid topic: $topicName5 in resource group $resourceGroupName with InputSchema $inputSchemaCloudEventSchema1"
+        $result = New-AzureRmEventGridTopic -ResourceGroup $resourceGroupName -Name $topicName5 -Location $location -InputSchema $inputSchemaCloudEventSchema1 -InputMappingField @{ id = "MyIdField"; topic = "MyTopicField"; eventtime = "MyEventTimeField"; subject = "MySubjectField"; eventtype = "MyEventTypeField" }
+        Assert-True {$false} "New-AzureRmEventGridTopic succeeded while it is expected to fail as inputmapping parameters is not null for input mapping schema CloudEventSchema"
+    }
+    catch
+    {
+        Assert-True {$true}
+    }
+
+    try
+    {
+        Write-Debug "Creating a fifth EventGrid topic: $topicName5 in resource group $resourceGroupName with InputSchema $inputSchemaCustomEventSchema1"
+        $result = New-AzureRmEventGridTopic -ResourceGroup $resourceGroupName -Name $topicName5 -Location $location -InputSchema $inputSchemaCustomEventSchema1
+        Assert-True {$false} "New-AzureRmEventGridTopic succeeded while it is expected to fail as inputmapping parameters are null"
+    }
+    catch
+    {
+        Assert-True {$true}
+    }
+
+    Write-Debug "Creating a fifth EventGrid topic: $topicName5 in resource group $resourceGroupName with InputSchema $inputSchemaCustomEventSchema1"
+    $result = New-AzureRmEventGridTopic -ResourceGroup $resourceGroupName -Name $topicName5 -Location $location -InputSchema $inputSchemaCustomEventSchema1 -InputMappingField @{ id = "MyIdField"; topic = "MyTopicField"; eventtime = "MyEventTimeField"; subject = "MySubjectField"; eventtype = "MyEventTypeField"; dataversion = "MyDataVersionField" }
+    Assert-True {$result.ProvisioningState -eq "Succeeded"}
+    Assert-True {$result.InputSchema -eq $expectedInputMappingCustomEventSchema} "InputSchema is not correct. CustomEventSchema is expected"
+
+    Write-Debug "Creating a sixth EventGrid topic: $topicName6 in resource group $resourceGroupName with InputSchema $inputSchemaCustomEventSchema2"
+    $result = New-AzureRmEventGridTopic -ResourceGroup $resourceGroupName -Name $topicName6 -Location $location -InputSchema $inputSchemaCustomEventSchema2 -InputMappingField @{ id = "MyIdField"; topic = "MyTopicField"; eventtime = "MyEventTimeField"; subject = "MySubjectField"; eventtype = "MyEventTypeField"; dataversion = "MyDataVersionField" } -InputMappingDefaultValue @{ subject = "MySubjectDefaultValue"; eventtype = "MyEventTypeDefaultValue"; dataversion = "MyDataVersionDefaultValue" }
+    Assert-True {$result.ProvisioningState -eq "Succeeded"}
+    Assert-True {$result.InputSchema -eq $expectedInputMappingCustomEventSchema} "InputSchema is not correct. CustomEventSchema is expected"
```

Comment:
Can you validate the input schema, input mapping field, and input default values? If that is not in the output, this is fine.

Reply:
We do validate the input mapping fields and default values if they are provided as part of the cmdlet execution. Please refer to ValidateInputMappingInfo() in NewAzureEventGridTopic.cs

Reply:
I meant to validate that these values are properly set when creating the EventGridTopic.  If that is not possible (if these values are not in the object output by the cmdlet), then this is fine.